core.js

```diff
@@ -757,24 +757,15 @@ exports.init = function (sbot, config) {
       () => {
         if (!state.has(feedId)) return cb()
 
-        self.query(
-          where(author(feedId)),
-          asOffsets(),
-          toCallback((err, offsets) => {
+        pull(
+          self.query(where(author(feedId)), asOffsets(), toPullStream()),
+          pull.asyncMap(log.del),
+          pull.collect((err) => {
```

Comment:
If you're not using the result from pull.collect, how about just using a pull.drain? pull.collect accumulates an array of all the results, in this case it'll be an array with thousands of `undefined` items. https://github.com/pull-stream/pull-stream/blob/29b4868bb3864c427c3988855c5d65ad5cb2cb1c/sinks/collect.js#L7

Could do like `pull.drain(noop, (err) => {`

Reply:
Actually, `pull.onEnd((err) => {` seems sweet https://github.com/pull-stream/pull-stream/blob/master/sinks/on-end.js

Reply:
```suggestion
          pull.onEnd((err) => {
```

Reply:
Thanks, but this is still around 100s versus 4s

Reply:
That might actually be really interesting to try in any case. I'm still trying to understand why this is so much slower, seems like a gotcha in other places as well if pull is this much slower than push.

Reply:
Yeah, would be good to learn what the actual perf problem is.

Reply:
Yeah. I'm doing this a bit on and off today. I'll report back when I find out more. Thanks for the feedback so far.

Reply:
so it turns out to be this one: https://github.com/ssbc/jitdb/issues/189. Maybe we should update jitdb to use a better default of changing it here?